lesson29/src/js/resetingpassword.js

```diff
@@ -0,0 +1,55 @@
+import { checkFormValidityInBlur } from "./modules/validation";
+import { togglePasswordDisplay } from "./modules/togglepassword";
+
+const passwordOfInput = document.querySelector(".js-form-password");
+const confirmPasswordOfInput = document.querySelector(".js-form-confirm-password");
+const currentPasswordOfInput = document.querySelector(".js-form-current-password");
+const formElements = [passwordOfInput, confirmPasswordOfInput, currentPasswordOfInput];
+const errorOfConfirmPassword = confirmPasswordOfInput.nextElementSibling;
+const eyeIcons = document.querySelectorAll(".js-eye-icon");
+const submitButton = document.querySelector(".js-submit-button");
+
+const isMatchPasswordFields = () => passwordOfInput.value === confirmPasswordOfInput.value;
+const isMatchRegisteredPassword = (userData) => currentPasswordOfInput.value === userData.password;
+const checkFormValidityToEnableSubmitButton = () => submitButton.disabled = passwordOfInput.value !== confirmPasswordOfInput.value;
+
+formElements.forEach(element => {
+    element.classList.add("invalid");
+
+    element.addEventListener("blur", () => {
+        checkFormValidityInBlur(submitButton, element);
+
+        if (passwordOfInput.value && confirmPasswordOfInput.value) errorOfConfirmPassword.textContent = isMatchPasswordFields() ? "" : "上記のPasswordと異なります。もう一度入力してください。";
+        if (document.getElementsByClassName("invalid").length === 0) {
+            checkFormValidityToEnableSubmitButton();
+        };
+    });
+});
+
+eyeIcons.forEach(icon => {
+    icon.addEventListener("click", (e) => togglePasswordDisplay(e.target))
```

Comment:
IMO
```suggestion
    icon.addEventListener("click", togglePasswordDisplay)
```
にして `togglePasswordDisplay`の内部からtargetを参照するとシンプルに書けます

Reply:
教えていただきありがとうございます！
知りませんでした！
eyeIconの取得方法と一緒に修正しました。
[7b07890](https://github.com/haru-programming/JavaScript-lessons/pull/40/commits/7b07890f8f08eb544a81cce35292012077b80372)